api/v1alpha1/inferencemodel_types.go

```diff
@@ -0,0 +1,168 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// InferenceModelSpec represents a specific model use case. This resource is
+// managed by the "Inference Workload Owner" persona.
+//
+// The Inference Workload Owner persona is: a team that trains, verifies, and
+// leverages a large language model from a model frontend, drives the lifecycle
+// and rollout of new versions of those models, and defines the specific
+// performance and latency goals for the model. These workloads are
+// expected to operate within an InferencePool sharing compute capacity with other
+// InferenceModels, defined by the Inference Platform Admin.
+//
+// InferenceModel's modelName (not the ObjectMeta name) is unique for a given InferencePool,
+// if the name is reused, an error will be shown on the status of a
+// InferenceModel that attempted to reuse. The oldest InferenceModel, based on
+// creation timestamp, will be selected to remain valid. In the event of a race
+// condition, one will be selected at random.
+type InferenceModelSpec struct {
+	// The name of the model as the users set in the "model" parameter in the requests.
+	// The name should be unique among the workloads that reference the same backend pool.
+	// This is the parameter that will be used to match the request with. In the future, we may
+	// allow to match on other request parameters. The other approach to support matching on
+	// on other request parameters is to use a different ModelName per HTTPFilter.
+	// Names can be reserved without implementing an actual model in the pool.
+	// This can be done by specifying a target model and setting the weight to zero,
+	// an error will be returned specifying that no valid target model is found.
+	//
+	// +optional
+	// +kubebuilder:validation:MaxLength=253
+	ModelName string `json:"modelName,omitempty"`
+	// Defines how important it is to serve the model compared to other models referencing the same pool.
+	//
+	// +optional
+	// +kubebuilder:default="Default"
+	Criticality *Criticality `json:"criticality,omitempty"`
+	// Allow multiple versions of a model for traffic splitting.
+	// If not specified, the target model name is defaulted to the modelName parameter.
+	// modelName is often in reference to a LoRA adapter.
+	//
+	// +optional
+	// +kubebuilder:validation:MinItems=1
```

Comment:
I'm not sure I follow - you can omit this but not have an empty list?

Reply:
Fair point. My intent was to just validate if there was an empty `targetModels:` left in a file for some reason. Removed the min items